CMakeLists.txt

```diff
@@ -38,43 +39,78 @@ set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${LIB_DIR})
 set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${LIB_DIR})
 
 # protobuf
-set (PROTOBUF_DIR ${THIRD_PARTY_DIR}/protobuf)
-set (PROTOBUF_INCLUDE ${PROTOBUF_DIR}/build/include)
-set (PROTOBUF_LIB ${PROTOBUF_DIR}/build/lib)
-set (PROTOC ${PROTOBUF_DIR}/build/bin/protoc)
-add_custom_target (protobuf_target)
-add_custom_command (TARGET protobuf_target
-    COMMAND ${CMAKE_COMMAND} -E echo "prepare protobuf library ..."
-    COMMAND /bin/sh -c "if [ ! -d '${PROTOBUF_DIR}/build' ]; then cd '${PROTOBUF_DIR}'; ./build.sh; fi"
-    COMMAND ${CMAKE_COMMAND} -E echo "protobuf done"
-    VERBATIM
-)
+find_package(Protobuf)
+if(Protobuf_FOUND)
```

Comment:
There is a space between the cmake keyword and the brackets in current CMakeList.txt (e.g. `option (...)`), we’d better use a unified format.